etl/pipelines/arkg_builder_pipeline.py

```diff
@@ -1,91 +1,106 @@
-from typing import override
+from typing import Annotated
 
+from pydantic import Field
 from pyoxigraph import Literal, NamedNode, Quad, Store
 
-from etl.models import WIKIPEDIA_BASE_URL
-from etl.models.types import AntiRecommendationKey, Iri, Predicate, RecordKey
-from etl.pipelines.kg_builder_pipeline import KgBuilderPipeline
+from etl.models import RDF_TYPE, WIKIPEDIA_BASE_URL, ArkgInstance, ArkgSchema
+from etl.models.types import AntiRecommendationKey, RecordKey
 
 
-class ArkgBuilderPipeline(KgBuilderPipeline):
+class ArkgBuilderPipeline:
     """
-    A concrete implementation of KgBuilderPipeline.
+    A pipeline to build Anti-Recommendation Knowledge Graphs.
 
-    Constructs a RDF store from a tuple of anti-recommendation graphs.
+    Constructs a RDF Store from a tuple of anti-recommendation graphs.
     """
 
-    def __init__(self, base_iri: Iri) -> None:
-        self.__base_iri = base_iri
-        self.__store = Store()
+    def __init__(self) -> None:
 
-    def __add_title_quad_to_store(self, record_key: RecordKey) -> None:
-        """
-        Add a RDF Quad with a `HAS_TITLE` predicate to the RDF Store.
+        self.__store = Store()
 
-        The subject of the Quad is the Store's base iri + record_key.
-        The object of the Quad is the record_key.
+    def __add_anti_recommendation_quads_to_store(
+        self,
+        *,
+        anti_recommendation_keys: tuple[AntiRecommendationKey, ...],
+        item_reviewed: Annotated[
+            str, Field(min_length=1, json_schema_extra={"strip_whitespace": True})
+        ],
+    ) -> None:
         """
+        Add `anti-recommendation` Quads to the RDF Store.
 
-        self.__store.add(
-            Quad(
-                NamedNode(self.__base_iri + record_key.replace(" ", "_")),
-                NamedNode(self.__base_iri + Predicate.HAS_TITLE.value),
-                Literal(record_key),
-            )
-        )
+        `item_reviewed` is the IRI of the item that is being anti-recommended.
 
-    def __add_url_quad_to_store(self, record_key: RecordKey) -> None:
+        Each anti_recommendation_key has 2 Quad expressions in this method:
+        - a `type` Quad that expresses the type of Review the anti_recommendation_key belongs to.
+        - an `item-reviewed` Quad that relates the anti_recommendation_key to item that is being anti-recommended.
         """
-        Add a RDF Quad with a `HAS_URL` predicate to the RDF Store.
+        for anti_recommendation_key in anti_recommendation_keys:
 
-        The subject of the Quad is the Store's base iri + record_key.
-        The object of the Quad is Wikipedia's base url + record_key.
-        """
+            anti_recommendation_instance = ArkgInstance.anti_recommendation_iri(
+                record_key=anti_recommendation_key.replace(" ", "_")
```

Comment:
Have the IRI factory method do the `.replace`, that shouldn't be the concern of this code.